### PR TITLE
GitHub Org Management

### PR DESCRIPTION
**Pull request by lelia**
# Description

This PR introduces declarative organization management to [OpenClarity](https://github.com/openclarity) using [Peribolos](https://github.com/uwu-tools/peribolos) and reusable [GitHub workflows](https://docs.github.com/en/actions/using-workflows).

A supplementary [README](https://github.com/openclarity/.github/blob/org-config-v1/org/README.md) has been created to cover the overall structure and usage details.

### CODE REVIEW

File: .github/CODEOWNERS
Comment (justaugustus) [inline .github/CODEOWNERS:1]: Let's pull these changes out into a separate PR (can combine in a PR with the other file changes I flagged).

File: .github/dependabot.yml
Comment (justaugustus) [inline .github/dependabot.yml:1]: Let's pull these changes out into a separate PR (can combine in a PR with the other file changes I flagged).

File: .github/stale.yml
Comment (justaugustus) [inline .github/stale.yml:1]: Let's pull these changes out into a separate PR (can combine in a PR with the other file changes I flagged).

File: .yamllint.yml
Comment (justaugustus) [inline .yamllint.yml:1]: Let's pull these changes out into a separate PR (can combine in a PR with the other file changes I flagged).

File: .github/workflows/test-sync.yml [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+      #     github-token-path: ./token
+      #     config-path: orgs
+      #     min-admins: 3
+      #     # TODO: Add service account name here
+      #     required-admins: "lelia,detiber,justaugustus"
+      #     fix-org: true
+      #     fix-org-members: true
+      #     fix-teams: true
+      #     fix-team-members: true
+      #     fix-team-repos: true
```
Comment (justaugustus) [inline .github/workflows/test-sync.yml:45]:
```suggestion
```

Let's opt to use Probot Settings for the per repo configurations.

**Review by justaugustus — CHANGES_REQUESTED** [review]
Great start, @lelia!

As next steps, I'd suggest two things:
1. I flagged a few file changes that I think can be safely pulled into a separate PR and merged.
2. Check out peribolos' `--dump` commands, so that you can get an [initial seed](https://github.com/uwu-tools/peribolos#initial-seed) of the org configuration, then break that up into the teams structure you've got laid out here

Okay, wait, two nits as well:
1. rename `orgs/` to `org/` or `config/`, etc., as this directory will only cover configuration for the `openclarity` org
2. let's opt to use probot settings GitHub App for the per repo configurations.

File: .github/CODEOWNERS
Comment (lelia) [reply .github/CODEOWNERS:1]: See https://github.com/openclarity/.github/pull/12

**Review by justaugustus — COMMENTED** [review]
https://github.com/openclarity/.github/pull/12 has merged, so those changes are safe to rebase out :)

**Review by justaugustus — CHANGES_REQUESTED** [review]
@lelia -- Oh, one more thing!
Please squash the existing commits on the next push :)

File: .github/workflows/dry-run.yml
[the remark this answers is not in the document]
Comment (lelia) [reply .github/workflows/dry-run.yml:1]: Agree that using a `dev` branch -> `main` sync workflow is probably the easiest way to work around this limitation and avoid the known issues with using `pull_request_target:` triggers. I'm going to test that branching model out in `/cisco-ospo` and port the workflow logic here once it's confirmed working.

Also going to file a ticket to investigate some solutions offered [here](https://securitylab.github.com/research/github-actions-preventing-pwn-requests/) and [here](https://blog.gitguardian.com/github-actions-security-cheat-sheet/) as a longer-term fix ✅

[the remark this answers is not in the document]
Comment (lelia) [reply .github/workflows/dry-run.yml:1]: Proof-of-concept example demonstrating the PR -> merge to `develop` -> dry run (`confirm: false`) workflow: https://github.com/cisco-ospo/.github/actions/runs/5733062747

A successful run of this job results in a prod run (`confirm: true`) being triggered after the code is synced from `develop` to `main`: https://github.com/cisco-ospo/.github/actions/runs/5733068217

File: org/README.md [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+
+### Peribolos Config
+
+The main configuration file is [org.yaml](./openclarity/org.yaml), with official documentation found [here](https://docs.prow.k8s.io/docs/components/cli-tools/peribolos/#org-configuration). This configuration is supported by supplementary, team-based configuration files stored in respectively-named subfolders.
+
+For instance, the VMClarity project has its configuration stored in a dedicated [teams.yaml](../org/openclarity/vmclarity/teams.yaml). When Peribolos runs, it scans the entire [org/](../org) directory tree for `.yaml` files, and can merge multiple config files together, if applicable.
+
+### GitHub Workflows
+
+Peribolos presently runs on a scheduled (hourly) and event-driven (merges to `main`) basis. These organization sync jobs are configured in [sync-org.yml](../.github/workflows/sync-org.yml). When new changes are proposed to the current configuration files stored in [org/](../org), a separate [dry-run.yml](../.github/workflows/dry-run.yml) job will run to prevent erroneous changes from being deployed.
```
[the remark this answers is not in the document]
Comment (lelia) [reply org/README.md:31]: ( Going to leave `README.md` untouched until we confirm the fork PR -> dry run on `develop` -> sync code to `main` -> prod run on `main` workflow is what we want to go with )

File: org/openclarity/chainclarity/teams.yaml
[the remark this answers is not in the document]
Comment (lelia) [reply org/openclarity/chainclarity/teams.yaml:1]: To that end, do we still need the `teams.yaml` file for `functionclarity`, given that's now publicly archived?

File: org/openclarity/functionclarity/teams.yaml [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+      - rmedvedo
+    privacy: closed
+  functionclarity-maintainers:
+    description: Maintainers for FunctionClarity repos
+    maintainers:
+      - naor-cisco
+    members:
+      - embonshai
+      - rmedvedo
+    privacy: closed
```
Comment (lelia) [inline org/openclarity/functionclarity/teams.yaml:18]: Restore chainclarity (for now)

File: .github/workflows/dry-run.yml [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+jobs:
+  lint-yaml:
+    uses: ./.github/workflows/_lint.yml
+  dry-run:
+    needs: lint-yaml
+    uses: ./.github/workflows/_sync.yml
+    permissions: read-all
+    secrets: inherit
+    with:
+      confirm: false
```
Comment (lelia) [inline .github/workflows/dry-run.yml:23]: Might make more sense to run these jobs in parallel

File: .github/workflows/dry-run.yml [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+  sync-branch:
+    needs: dry-run
+    runs-on: ubuntu-latest
+    steps:
+      - name: 🔄 sync branch
+        uses: connor-baer/action-sync-branch@v1.0.3
+        with:
+          token: ${{ secrets.SERVICE_ACCOUNT_TOKEN }}
+          branch: main
+          force: false
```
Comment (lelia) [inline .github/workflows/dry-run.yml:33]: Might not need this if we can run validation workflows (which could include linting and a dry-run) on fork PRs _once they've been approved_

File: org/README.md [hunk 6]
```diff
@@ -0,0 +1,47 @@
...
+.
+├── .github # GitHub repo config
+│   └── workflows # GitHub workflow files
+│       ├── dry-run.yml # Dry-run of Peribolos
+│       ├── dump-config.yml # Dump current org config
+│       └── sync-org.yml # Sync orgs with Peribolos
+└── org # Peribolos config directory
+    ├── README.md # This README :)
+    └── openclarity # Parent folder for org structure
+        └── org.yaml # Main Peribolos configuration file
```
Comment (lelia) [inline org/README.md:20]: Adjust to reflect new reusable workflow structure

Comment (lelia) [reply org/README.md]: Done

File: org/openclarity/chainclarity/teams.yaml [hunk 7]
```diff
@@ -0,0 +1,18 @@
+---
```
Comment (FrimIdan) [inline org/openclarity/chainclarity/teams.yaml:1]: @lelia I think you confused `chainclarity` (that was a PoC repo that is on hold right now and it's private) with `functionclarity` (that is Public archive)

chainclarity admins are
image
and maintainers are
image

Comment (lelia) [reply org/openclarity/chainclarity/teams.yaml:1]: Good catch, @FrimIdan ! That was definitely a mis-paste from the `functionclarity` clipboard data 🙂 

I'll push the updated membership affiliations on the next commit.

File: org/openclarity/org.yaml [hunk 8]
```diff
@@ -0,0 +1,82 @@
...
+members_can_create_repositories: false
+name: OpenClarity
+teams:
+  org-admins:
+    description: Admins for OpenClarity repos
+    maintainers:
+      - detiber
+      - edwarnicke
+      - justaugustus
+      - lelia
```
Comment (detiber) [inline org/openclarity/org.yaml:76]:
```suggestion
      - lelia
      - natalisucks
```

**Review by openclarity-service — APPROVED** [review]

**Review by justaugustus — APPROVED** [review]
Great work on this, @lelia!

**Pull request merged**